src/Views/SilenceForm/Updates.elm

```diff
@@ -32,119 +37,140 @@ update msg model =
-
-        UpdateMatcherRegex silence matcher bool ->
-            let
-                matchers =
-                    Utils.List.replaceIf (\x -> x == matcher) { matcher | isRegex = bool } silence.matchers
-            in
-                ( { model | silence = Success { silence | matchers = matchers } }, Cmd.none )
-
-        NewDefaultTimeRange time ->
-            let
-                startsAt =
-                    Utils.Date.fromTime time
-
-                duration =
-                    Utils.Date.duration (2 * Time.hour)
-
-                endsAt =
-                    Utils.Date.fromTime (time + 2 * Time.hour)
-
-                sil =
-                    case model.silence of
-                        Success s ->
-                            s
-
-                        _ ->
-                            nullSilence
-            in
-                ( { model | silence = Success { sil | startsAt = startsAt, duration = duration, endsAt = endsAt } }, Cmd.none )
-
-        FetchSilence silenceId ->
-            ( { model | silence = model.silence }, Silences.Api.getSilence silenceId (SilenceFetch >> MsgForSilenceForm) )
+                    ( model, Navigation.newUrl "/#/silence" )
 
         NewSilence ->
-            ( { model | silence = model.silence }, Cmd.map MsgForSilenceForm (Task.perform NewDefaultTimeRange Time.now) )
+            ( model, Task.perform (NewDefaultTimeRange >> UpdateField >> MsgForSilenceForm) Time.now )
 
-        SilenceFetch sil ->
+        FetchSilence silenceId ->
+            ( model, Silences.Api.getSilence silenceId (SilenceFetch >> MsgForSilenceForm) )
+
+        SilenceFetch (Success silence) ->
+            ( { model | form = fromSilence silence, silence = Ok silence }
+            , Task.perform (PreviewSilence >> MsgForSilenceForm) (Task.succeed silence)
+            )
+
+        SilenceFetch _ ->
+            ( model, Cmd.none )
+
+        PreviewSilence silence ->
+            ( { model | silence = Ok { silence | silencedAlertGroups = Loading } }
+            , Alerts.Api.alertGroups
+                { nullFilter | text = Just (Utils.List.mjoin silence.matchers) }
+                |> Cmd.map (AlertGroupsPreview >> MsgForSilenceForm)
+            )
+
+        AlertGroupsPreview alertGroups ->
+            case model.silence of
+                Ok sil ->
+                    ( { model | silence = Ok { sil | silencedAlertGroups = alertGroups } }
+                    , Cmd.none
+                    )
+
+                Err _ ->
+                    ( model, Cmd.none )
+
+        UpdateField fieldMsg ->
             let
-                cmd =
-                    case sil of
-                        Success sil ->
-                            Task.perform identity (Task.succeed (PreviewSilence sil))
+                newForm =
+                    updateForm fieldMsg model.form
 
-                        _ ->
-                            Cmd.none
+                newSilence =
```

Comment:
I find it easier to keep it there, and then use in views, because I need it to enable/disable the button depending on the validation state, and initiate create message with `silence`.